CONTRIBUTING.rst

```diff
@@ -256,24 +263,68 @@ When developing, you can make Python changes available in R by running (from mlf
 Please also follow the recommendations from the
 `Advanced R - Style Guide <http://adv-r.had.co.nz/Style.html>`_ regarding naming and styling.
 
+Java
+~~~~
+
+If contributing to MLflow's Java APIs or modifying Java documentation,
+install `Java <https://www.java.com/>`_ and `Apache Maven <https://maven.apache.org/download.cgi>`_.
+
+Certain MLflow modules are implemented in Java, under the ``mlflow/java/`` directory.
+These are the Java Tracking API client (``mlflow/java/client``) and the Model Scoring Server
+for Java-based models like MLeap (``mlflow/java/scoring``).
+
+Other Java functionality (like artifact storage) depends on the Python package, so first install
+the Python package in a conda environment as described above.
+`Install <https://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+the Java 8 JDK (or above), and `download <https://maven.apache.org/download.cgi>`_
+and `install <https://maven.apache.org/install.html>`_ Maven. You can then build and run tests via:
+
+.. code-block:: bash
+
+  cd mlflow/java
+  mvn compile test
+
+If opening a PR that makes API changes, please regenerate API documentation as described in
+`Writing Docs`_ and commit the updated docs to your PR branch.
+
 Python
 ~~~~~~
+If you are contributing in Python, make sure that you have satisfied all the `General prerequisites and dependencies`_,
+including installing ``pytest``, as you will need it for the sections described below.
+
+Writing Python Tests
+++++++++++++++++++++
+If your PR includes code that isn't currently covered by our tests (e.g. adding a new flavor, adding
+autolog support to a flavor, etc.), you should write tests that cover your new code. MLflow currently
+uses ``pytest==3.2.1`` for testing. Your tests should be added to the relevant file under ``tests``, or
+if there is no appropriate file, in a new file prefixed with ``test_`` so that ``pytest`` includes that
+file for testing.
+
+If your tests require usage of a tracking URI, the
+`pytest fixture <https://docs.pytest.org/en/3.2.1/fixture.html>`_
+`tracking_uri_mock <https://github.com/mlflow/mlflow/blob/master/tests/conftest.py#L74>`_ is automatically set up
+for every tests. It sets up a mock tracking URI that will set itself up before your test runs and tear itself down after.
+If you want to deactivate the mock for your test, mark the test with `@pytest.mark.notrackingurimock` operator.
+
+Running Python Tests
+++++++++++++++++++++
+
 Verify that the unit tests & linter pass before submitting a pull request by running:
 
 .. code-block:: bash
 
     ./lint.sh
-    ./travis/run-small-python-tests.sh
+    ./dev/run-small-python-tests.sh
     # Optionally, run large tests as well. Travis will run large tests on your pull request once
```

Comment:
I dont think travis runs the tests anymore. Github actions for the win!
```suggestion
    # Optionally, run large tests as well. Github actions will run large tests on your pull request once
```

Reply:
There places where we will have to change all occurrences of how Travis is replaced with Github's actions.